### DISCUSSION TITLE
Display center point of Hexagon layer

### DISCUSSION THREAD

**Discussion by MTVetter** [visgl/deck.gl#5259] [Q&A]
Is there any way to symbolize the hexagons in a hexagon layer like the image below?
![image](https://user-images.githubusercontent.com/21963735/102251969-1bc34f80-3ecb-11eb-9c9d-bc414daa8d36.png)

If it's not possible to do something like that is there a way to possibly create a scatterplot layer using the center of the hexagon?

**Comment by Pessimistress** [comment]
You can supply your own sub layer class to `HexagonLayer`, something like:

```js
class MyCustomLayer extends CompositeLayer {
  renderLayers() {
    return [
      new ColumnLayer(this.props),
      new ScatterplotLayer(
        this.props,
        this.getSubLayerProps({
          id: 'center',
          getRadius: 100,
          stroked: false,
          filled: true,
          getFillColor: [0, 0, 0]
        })
      )
    ]
  }
}

new HexagonLayer({
  // hexagon layer props...
  _subLayerProps: {
    'hexagon-cell': {
      type: MyCustomLayer
    }
  }
})
```

Reference
- [CompositeLayer _subLayerProps](https://deck.gl/docs/api-reference/core/composite-layer#_sublayerprops)
- [HexagonLayer sub layers](https://deck.gl/docs/api-reference/aggregation-layers/hexagon-layer#sub-layers)
- [Writing a composite layer](https://deck.gl/docs/developer-guide/custom-layers/composite-layers)
